server/jetstream_test.go

```diff
@@ -15209,6 +15209,51 @@ func TestJetStreamPullConsumerHeartBeats(t *testing.T) {
 	}
 }
 
+func TestJetStreamRecoverStreamWithDeletedMessagesNonCleanShutdown(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer s.Shutdown()
+
+	if config := s.JetStreamConfig(); config != nil {
+		defer removeDir(t, config.StoreDir)
+	}
+
+	nc, js := jsClientConnect(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{Name: "T"})
+	require_NoError(t, err)
+
+	for i := 0; i < 100; i++ {
+		js.Publish("T", []byte("OK"))
+	}
+
+	js.DeleteMsg("T", 22)
+
+	// Now we need a non-clean shutdown.
+	// For this use case that means we do *not* write the fss file.
+	sd := s.JetStreamConfig().StoreDir
+	fss := path.Join(sd, "$G", "streams", "T", "msgs", "1.fss")
+
+	// Stop current
+	nc.Close()
+	s.Shutdown()
+
+	// Remove fss file to simulate a non-clean shutdown.
+	err = os.Remove(fss)
+	require_NoError(t, err)
+
+	// Restart.
+	s = RunJetStreamServerOnPort(-1, sd)
+	defer s.Shutdown()
+
+	nc, js = jsClientConnect(t, s)
+	defer nc.Close()
+
+	// Make sure we recovered our stream
+	_, err = js.StreamInfo("T")
+	require_NoError(t, err)
```

Comment:
check for 99 messages and that `js.GetMsg("T", 22)` fails?

Am I right to assume that that upon recovery the missing fss file should exist again?

Reply:
We have hundreds of tests that already do this. I understand where you are coming from and actually started to write that and was like, the stream info message count deleted messages have tests everywhere for over a year. So I say no.

Reply:
Meaning I do not believe it buys us anything over the current tests that are in place.

Reply:
The bug was in rebuilding the fss state. That file does not actually exists during the server running, only on exit as a short cut on restart.

Reply:
Also just as an fyi, I always write the test to fail first and it did with "stream not found", which is what was reported, so as long as the fix had the stream being found I think we are good, IMO.